CMakeLists.txt

```diff
@@ -142,22 +145,67 @@ add_executable(${PROJECT_NAME} ${GUI_TYPE}
                ${HEADERS}
                ${FORMS}
                ${RESOURCES}
-               ${CMAKE_SOURCE_DIR}/res/windows/resources.rc)
-
-target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} LibREDasm QHexEdit)
-add_dependencies(${PROJECT_NAME} LibREDasm)
-
-if(WIN32)
-    target_link_libraries(${PROJECT_NAME} Qt5::Core Qt5::Gui Qt5::Widgets LibREDasm)
-else()
-    target_link_libraries(${PROJECT_NAME} Qt5::Core Qt5::Gui Qt5::Widgets pthread LibREDasm)
-endif()
+               ${CMAKE_SOURCE_DIR}/res/windows/resources.rc
+)
+
+# TODO: refactor this to target_include_directories(target PUBLIC ...) in the submodules
+target_include_directories(${PROJECT_NAME} PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}
+        LibREDasm
+        QHexEdit
+)
+
+target_link_libraries(${PROJECT_NAME} PRIVATE
+    Qt5::Core
+    Qt5::Gui
+    Qt5::Widgets
+    LibREDasm
+)
+
+# Include Threads
+# TODO: perhaps this should be a PUBLIC dependency of LibREDasm?
```

Comment:
This and the other `TODO` require some refactors of the `LibREDasm` CMakeLists.txt. I did not do this because I couldn't get everything to compile on my machine.